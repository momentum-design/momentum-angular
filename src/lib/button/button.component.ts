import {
  Component,
  Input,
  ViewChild,
  ElementRef,
  OnInit,
  HostBinding,
  AfterViewInit,
  OnChanges,
  AfterContentInit,
} from '@angular/core';

@Component({
  selector: 'button[cui-button], a[cui-button], input[cui-button]',
  exportAs: 'cuiButton',
  template: `
    <div #divBody>
      <cui-loading *ngIf="loading"></cui-loading>
      <span [ngStyle]="{ opacity: opacity }">
        <ng-content></ng-content>
      </span>
    </div>
  `,
})
export class ButtonComponent
  implements OnInit, AfterViewInit, OnChanges, AfterContentInit {
  /** @option Active Sets button to active | false */
  @Input() public active: boolean = false;
  /** @option AriaLabel Text to display for blindness accessibility features | '' */
  /** @option AriaLabelledBy ID to reference for blindness accessibility feature | '' */
  /** @option Creates a circle shaped button | false */
  @Input() public circle: boolean = false;
  /** @option Optional css class string | '' */
  @Input() public class: string = '';
  /** @option Sets optional button color | '' */
  @Input() public color: string = '';
  /** @option Sets the attribute disabled to Button | false */
  @Input() public disabled: boolean = false;
  /** @option Sets expand css styling to widen the Button | false */
  @Input() public expand: boolean = false;
  /** @option Sets href for an anchor tagged button | '' */
  @Input() public href: string = '';
  /** @option Activates the loading animation and sets the button to disabled | false */
  @Input() public loading: boolean = false;
  /** @option Boolean to remove Button's default style | false */
  @Input() public removeStyle: boolean = false;
  /** @option Size value | '36' */
  @Input() public size: string = '36';
  /** @option Html button type | 'button' */
  @Input() public type: string = 'button';

  @ViewChild('divBody') element: ElementRef;
  @ViewChild('loader') loaderChild: ElementRef;

  @HostBinding('attr.width') public width;
  @HostBinding('attr.alt') alt;
  @HostBinding('attr.active') get activeOption() {
    return this.active || null;
  }
  @HostBinding('attr.disabled') get disabledOption() {
    return this.disabled || null;
  }
  @HostBinding('attr.href') get hrefOption() {
    return this.href || null;
  }
  @HostBinding('attr.type') get typeOption() {
    return this.type || 'button';
  }

  @HostBinding('class') get className(): string {
    return (
      'cui-button' +
      `${(this.circle && ` cui-button--circle`) || ''}` +
      `${(this.removeStyle && ' cui-button--none') || ''}` +
      `${(this.getSize() && ` cui-button--${this.getSize()}`) || ''}` +
      `${(this.getColor() && ` cui-button--${this.getColor()}`) || ''}` +
      `${(this.expand && ` cui-button--expand`) || ''}` +
      `${(this.class && ` ${this.class}`) || ''}` +
      `${(this.disabled && ` cui-button--disabled`) || ''}` +
      `${(this.active && !this.disabled && ` active`) || ''}` +
      ``
    );
  }

  public initWidth: string;

  constructor(private el: ElementRef) {}

  ngOnInit() {
    this.disabled = this.disabled || this.loading || null;
  }

  ngAfterViewInit() {
    this.initWidth = `${this.element.nativeElement.offsetWidth + 0.001}px`;

    if (
      this.type &&
      this.type !== 'button' &&
      this.type !== 'reset' &&
      this.type !== 'submit'
    ) {
      throw new Error(
        'cui-button: Button type must be one of the following: button, reset, submit'
      );
    }
  }

  ngAfterContentInit() {
    setTimeout(() => this.setAriaLabel(), 100);
  }

  ngOnChanges(changes) {
    if (changes.loading && changes.loading.currentValue) {
      this.width = this.initWidth;
    } else {
      this.width = '';
    }
  }

  public get opacity(): number {
    return this.loading ? 0 : 1;
  }

  private getColor = () => {
    if (this.removeStyle) {
      return false;
    }
    return this.color === 'none' ? 'color-none' : this.color;
  }

  private getSize = () => {
    if (this.removeStyle) {
      return false;
    }
    const validButtonSize = this.checkButtonSize();

    if (!this.circle && !validButtonSize) {
      console.warn(
        `[@collab-ui/angular] Button: selected size is not supported
         for non-circular button. Size will default to 36`
      );

      return '36';
    } else {
      return this.size === 'none' ? 'size-none' : this.size;
    }
  }

  private checkButtonSize = () =>
    ['none', '28', '36', '40', '52', 28, 36, 40, 52].includes(this.size)

  private setAriaLabel() {
    let ariaLabel = this.getAttr('aria-label');
    if (!ariaLabel) {
      if (this.element.nativeElement.children.length > 0) {
        throw new Error(
          'cui-button: content is not a string, you must add an "ariaLabel" for accessibility.'
        );
      } else {
        ariaLabel = this.element.nativeElement.innerText;
      }
    }
  }

  private getAttr(attr) {
    return this.el.nativeElement.getAttribute(attr);
  }
}

/**
* @component button
* @section active
* @angular
*
<div>
  <button cui-button aria-label="myAriaLabel" [active]=true (click)="onClick()">Test Me</button>
  <button cui-button aria-label="myAriaLabel" [active]=false (click)="onClick()">Test Me</button>

  <a cui-button aria-label="myAriaLabel" [active]=true>Test Me</a>
  <a cui-button aria-label="myAriaLabel" [active]=false>Test Me</a>
</div>
*/

/**
 * @component button
 * @section default
 * @angular
 *
<div>
  <button cui-button aria-label="myAriaLabel" (click)="onClick()">Test Me</button>
</div>
 */

/**
 * @component button
 * @section tags
 * @angular
 *
<div>
  <button cui-button aria-label="myAriaLabel" (click)="onClick()">Test Me</button>
  <a cui-button aria-label="myAriaLabel" (click)="onClick()">Test Me</a>
  <input cui-button aria-label="myAriaLabel" (click)="onClick()" />
</div>
 */

/**
 * @component button
 * @section color
 * @angular
 *
<div class='row'>
  <button cui-button aria-label="myAriaLabel" color="blue" (click)="onClick()">Test Me</button>
  <button cui-button aria-label="myAriaLabel" [color]="myColorVariable" (click)="onClick()">Test Me</button>
</div>
 */

/**
 * @component button
 * @section size
 * @angular
 *
<div>
  <button cui-button aria-label="myAriaLabel" size="none" (click)="onClick()">None</button>
  <button cui-button aria-label="myAriaLabel" size="28" (click)="onClick()">size 28</button>
  <button cui-button aria-label="myAriaLabel" size="36" (click)="onClick()">size 36</button>
  <button cui-button aria-label="myAriaLabel" size="40" (click)="onClick()">size 40</button>
  <button cui-button aria-label="myAriaLabel" size="52" (click)="onClick()">size 52</button>
  <button cui-button aria-label="myAriaLabel" [size]="mySizeVariable" (click)="onClick()">sizeVariable</button>

</div>

/**
 * @component button
 * @section circle
 * @angular
 *
<div>
  <button cui-button aria-label="myAriaLabel" [circle]=true (click)="onClick()">C</button>
  <button cui-button aria-label="myAriaLabel" [circle]=false (click)="onClick()">C</button>
  <button cui-button aria-label="myAriaLabel" [circle]="myCircleVariable" (click)="onClick()">Var</button>
</div>
 */

/**
 * @component button
 * @section expanded
 * @angular
 *
<div>
  <button cui-button aria-label="myAriaLabel" [expand]=true (click)="onClick()">Test Me</button>
</div>
 */

/**
 * @component button
 * @section disabled
 * @angular
 *
<div>
  <button cui-button aria-label="myAriaLabel" [disabled]=true (click)="onClick()">Test Me</button>
  <button cui-button aria-label="myAriaLabel" [disabled]=false (click)="onClick()">Test Me</button>

  <a cui-button aria-label="myAriaLabel" [disabled]=true>Test Me</a>
  <a cui-button aria-label="myAriaLabel" [disabled]=false>Test Me</a>
</div>
 */

/**
 * @component button
 * @section loading
 * @angular
 *
<div>
  <button cui-button [attr.aria-label]="myAriaLabelVar" [loading]=true (click)="onClick()">Test me</button>
  <button cui-button aria-label="myAriaLabel" [loading]=false (click)="onClick()">Test me</button>
</div>
*/

/**
 * @component button
 * @section type
 * @angular
 *
<div>
  <button cui-button [attr.aria-label]="myAriaLabelVar" type="button" (click)="onClick()">Test me</button>
  <button cui-button aria-label="myAriaLabel" [type]="myTypeVariable" (click)="onClick()">Test me</button>
</div>
*/
