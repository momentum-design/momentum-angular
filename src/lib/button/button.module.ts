import { NgModule, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ButtonComponent } from './button.component';
import { LoadingModule } from '../loading';

@NgModule({
  imports: [
    CommonModule,
    LoadingModule.forRoot(),
  ],
  exports: [ButtonComponent],
  declarations: [ButtonComponent],
})
export class ButtonModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: ButtonModule,
      providers: [],
    };
  }
}
