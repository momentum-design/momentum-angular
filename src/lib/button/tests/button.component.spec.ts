import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { CUSTOM_ELEMENTS_SCHEMA, Component } from '@angular/core';
import { ButtonModule } from '../button.module';

// tslint:disable:no-use-before-declare
describe('ButtonComponent', () => {
  let testComponent: TestAppComponent;
  let testFixture: ComponentFixture<TestAppComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [TestAppComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
      imports: [ButtonModule],
    }).compileComponents();
  }));

  beforeEach(() => {
    testFixture = TestBed.createComponent(TestAppComponent);
    testComponent = testFixture.debugElement.componentInstance;
  });

  describe('of type Button', () => {
    it(`environment's should be created successfully`, () => {
      testComponent.ariaLabel = 'aNewAriaLabel';
      testFixture.detectChanges();
      expect(testComponent).toBeTruthy();
    });

    it('should render as a button html element with required parameters', () => {
      testComponent.ariaLabel = 'aNewAriaLabel';
      testComponent.ariaLabelledBy = 'aNewArialabelledBy';
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const button = buttonNativeElement.querySelector('button');

      expect(button).not.toBeNull();
      expect(button).toMatchSnapshot();
      expect(button.className).toContain('cui-button cui-button--36');
    });

    it('should render the ariaLabel option', () => {
      testComponent.ariaLabel = 'aNewAriaLabel';
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const button = buttonNativeElement.querySelector('button');
      expect(button.getAttribute('aria-label')).toEqual(
        testComponent.ariaLabel
      );
    });

    it('should render the ariaLabelledBy option', () => {
      testComponent.ariaLabel = 'aNewAriaLabel';
      testComponent.ariaLabelledBy = 'aNewArialabelledBy';
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const button = buttonNativeElement.querySelector('button');

      expect(button.getAttribute('aria-labelledby')).toEqual(
        testComponent.ariaLabelledBy
      );
    });

    it('should render the LoaderComponent and disable button if the loading option is true', () => {
      testComponent.ariaLabel = 'ariaLabel';
      testComponent.loading = true;
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const button = buttonNativeElement.querySelector('button');

      expect(button).not.toBeNull();
      expect(button.className).toContain('cui-button cui-button--36');

      const loader = button.querySelector('CUI-LOADING');
      expect(loader).not.toBeNull();

      expect(button.getAttribute('disabled')).toBe('true');
      expect(button.className).toContain('cui-button--disabled');
    });

    it('should not render the LoaderComponent if loading option is absent or false', () => {
      testComponent.ariaLabel = 'ariaLabel';
      testFixture.detectChanges();
      let buttonNativeElement = testFixture.nativeElement;
      let button = buttonNativeElement.querySelector('button');
      let loader = button.querySelector('CUI-LOADING');
      expect(loader).toBeNull();

      testComponent.loading = false;
      testFixture.detectChanges();
      buttonNativeElement = testFixture.nativeElement;
      button = buttonNativeElement.querySelector('button');
      loader = button.querySelector('CUI-LOADING');
      expect(loader).toBeNull();
    });

    it('should remove all styles with removeStyle option', () => {
      testComponent.ariaLabel = 'ariaLabel';
      testComponent.removeStyle = true;
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const button = buttonNativeElement.querySelector('button');
      expect(button.className).toContain('cui-button--none');
    });

    it('should ignore color or size with removeStyle option', () => {
      testComponent.ariaLabel = 'ariaLabel';
      testComponent.color = 'blue';
      testComponent.size = '40';
      testComponent.removeStyle = true;
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const button = buttonNativeElement.querySelector('button');
      expect(button.className).toContain('cui-button--none');
      expect(button.className).not.toContain('cui-button--blue');
      expect(button.className).not.toContain('cui-button--40');
    });

    it('should render with correct className if the color option is none', () => {
      testComponent.ariaLabel = 'ariaLabel';
      testComponent.color = 'none';
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const button = buttonNativeElement.querySelector('button');
      expect(button.className).toContain('cui-button--color-none');
    });

    it('should be of type button by default', () => {
      testComponent.ariaLabel = 'ariaLabel';
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const button = buttonNativeElement.querySelector('button');
      expect(button.getAttribute('type')).toEqual('button');
    });

    it('should show active class when passed the active option', () => {
      testComponent.active = true;
      testComponent.ariaLabel = 'aNewAriaLabel';
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const button = buttonNativeElement.querySelector('button');
      expect(button.getAttribute('active')).toBe('true');
      expect(button.className).toContain('active');
    });

    it('should render the type when type option is defined', () => {
      testComponent.type = 'submit';
      testComponent.ariaLabel = 'aNewAriaLabel';
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const button = buttonNativeElement.querySelector('button');
      expect(button.getAttribute('type')).toEqual('submit');
    });

    it('should render additional classNames from the className option', () => {
      testComponent.class = 'extraClassName';
      testComponent.ariaLabel = 'aNewAriaLabel';
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const button = buttonNativeElement.querySelector('button');

      expect(button.className).toContain(testComponent.class);
    });

    it('should render as disabled if the disabled option is defined', () => {
      testComponent.disabled = true;
      testComponent.ariaLabel = 'aNewAriaLabel';
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const button = buttonNativeElement.querySelector('button');

      expect(button.getAttribute('disabled')).toBe('true');
      expect(button.className).toContain('cui-button--disabled');
    });

    it('should render as an expanded button if the expand option is defined', () => {
      testComponent.expand = true;
      testComponent.ariaLabel = 'aNewAriaLabel';
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const button = buttonNativeElement.querySelector('button');

      expect(button.className).toContain('cui-button--expand');
    });

    it('should handle onClick event when click option is defined', () => {
      testComponent.ariaLabel = 'aNewAriaLabel';
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const button = buttonNativeElement.querySelector('button');

      button.click();
      expect(testComponent.clickCount).toBe(1);
    });

    describe('should render the correct size', () => {
      beforeEach(() => {
        testComponent.ariaLabel = 'aNewAriaLabel';
      });

      it('when the circle option is true and size hasn\'t been set', () => {
        testComponent.circle = true;
        testFixture.detectChanges();

        const buttonNativeElement = testFixture.nativeElement;
        const button = buttonNativeElement.querySelector('button');

        expect(button.className).toContain('cui-button--36');
      });

      it('when the size option is not valid', () => {
        spyOn(console, 'warn');
        testComponent.size = '5000';
        testFixture.detectChanges();

        const buttonNativeElement = testFixture.nativeElement;
        const button = buttonNativeElement.querySelector('button');

        expect(button.className).toContain('cui-button--36');
        expect(console.warn).toHaveBeenCalled();
      });

      it('when the size option isn`t specified', () => {
        testFixture.detectChanges();

        const buttonNativeElement = testFixture.nativeElement;
        const button = buttonNativeElement.querySelector('button');
        expect(button.className).toContain('cui-button--36');
      });

      it('when the size option is defined as none', () => {
        testComponent.size = 'none';
        testFixture.detectChanges();

        const buttonNativeElement = testFixture.nativeElement;
        const button = buttonNativeElement.querySelector('button');
        expect(button.className).toContain('cui-button--size-none');
      });

      it('when the size option is set to 28', () => {
        testComponent.size = '28';
        testFixture.detectChanges();

        const buttonNativeElement = testFixture.nativeElement;
        const button = buttonNativeElement.querySelector('button');
        expect(button.className).toContain('cui-button--28');
      });

      it('when the size option is set to 40', () => {
        testComponent.size = '40';
        testFixture.detectChanges();

        const buttonNativeElement = testFixture.nativeElement;
        const button = buttonNativeElement.querySelector('button');
        expect(button.className).toContain('cui-button--40');
      });

      it('when the size option is set to 52', () => {
        testComponent.size = '52';
        testFixture.detectChanges();

        const buttonNativeElement = testFixture.nativeElement;
        const button = buttonNativeElement.querySelector('button');
        expect(button.className).toContain('cui-button--52');
      });

      it('when the size option is set to 20 and is a circle', () => {
        testComponent.circle = true;
        testComponent.size = '20';
        testFixture.detectChanges();

        const buttonNativeElement = testFixture.nativeElement;
        const button = buttonNativeElement.querySelector('button');
        expect(button.className).toContain('cui-button--circle');
        expect(button.className).toContain('cui-button--20');
      });

      it('when the size option is set to 32 and is a circle', () => {
        testComponent.circle = true;
        testComponent.size = '32';
        testFixture.detectChanges();

        const buttonNativeElement = testFixture.nativeElement;
        const button = buttonNativeElement.querySelector('button');
        expect(button.className).toContain('cui-button--circle');
        expect(button.className).toContain('cui-button--32');
      });

      it('when the size option is set to 44 and is a circle', () => {
        testComponent.circle = true;
        testComponent.size = '44';
        testFixture.detectChanges();

        const buttonNativeElement = testFixture.nativeElement;
        const button = buttonNativeElement.querySelector('button');
        expect(button.className).toContain('cui-button--circle');
        expect(button.className).toContain('cui-button--44');
      });

      it('when the size option is set to 56 and is a circle', () => {
        testComponent.circle = true;
        testComponent.size = '56';
        testFixture.detectChanges();

        const buttonNativeElement = testFixture.nativeElement;
        const button = buttonNativeElement.querySelector('button');
        expect(button.className).toContain('cui-button--circle');
        expect(button.className).toContain('cui-button--56');
      });

      it('when the size option is set to 68 and is a circle', () => {
        testComponent.circle = true;
        testComponent.size = '68';
        testFixture.detectChanges();

        const buttonNativeElement = testFixture.nativeElement;
        const button = buttonNativeElement.querySelector('button');
        expect(button.className).toContain('cui-button--circle');
        expect(button.className).toContain('cui-button--68');
      });

      it('when the size option is set to 72 and is a circle', () => {
        testComponent.circle = true;
        testComponent.size = '72';
        testFixture.detectChanges();

        const buttonNativeElement = testFixture.nativeElement;
        const button = buttonNativeElement.querySelector('button');
        expect(button.className).toContain('cui-button--circle');
        expect(button.className).toContain('cui-button--72');
      });

      it('when the size option is set to 84 and is a circle', () => {
        testComponent.circle = true;
        testComponent.size = '84';
        testFixture.detectChanges();

        const buttonNativeElement = testFixture.nativeElement;
        const button = buttonNativeElement.querySelector('button');
        expect(button.className).toContain('cui-button--circle');
        expect(button.className).toContain('cui-button--84');
      });
    });
  });

  describe('of type Anchor', () => {
    it('should render as an anchor html element with required parameters', () => {
      testComponent.ariaLabel = 'aNewAriaLabel';
      testComponent.ariaLabelledBy = 'aNewArialabelledBy';
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const anchor = buttonNativeElement.querySelector('a');

      expect(anchor).not.toBeNull();
      expect(anchor).toMatchSnapshot();
      expect(anchor.className).toContain('cui-button cui-button--36');
    });

    it('should render the LoaderComponent if the loading option is true', () => {
      testComponent.ariaLabel = 'ariaLabel';
      testComponent.loading = true;
      testFixture.detectChanges();
      expect(testComponent).toBeTruthy();

      const buttonNativeElement = testFixture.nativeElement;
      const anchor = buttonNativeElement.querySelector('a');

      const loader = anchor.querySelector('CUI-LOADING');
      expect(loader).not.toBeNull();
    });

    it('should load with provided href option', () => {
      testComponent.ariaLabel = 'ariaLabel';
      testComponent.href = 'www.google.com';
      testFixture.detectChanges();
      expect(testComponent).toBeTruthy();

      const buttonNativeElement = testFixture.nativeElement;
      const anchor = buttonNativeElement.querySelector('a');

      expect(anchor.getAttribute('href')).toEqual(testComponent.href);
    });
  });

  describe('of type Input', () => {
    it('should render as an anchor html element with required parameters', () => {
      testComponent.ariaLabel = 'aNewAriaLabel';
      testComponent.ariaLabelledBy = 'aNewArialabelledBy';
      testFixture.detectChanges();

      const buttonNativeElement = testFixture.nativeElement;
      const input = buttonNativeElement.querySelector('input');

      expect(input).not.toBeNull();
      expect(input).toMatchSnapshot();
      expect(input.className).toContain('cui-button cui-button--36');
    });

    it('should render the LoaderComponent if the loading option is true', () => {
      testComponent.ariaLabel = 'ariaLabel';
      testComponent.loading = true;
      testFixture.detectChanges();
      expect(testComponent).toBeTruthy();
      const buttonNativeElement = testFixture.nativeElement;
      const input = buttonNativeElement.querySelector('input');
      const loader = input.querySelector('CUI-LOADING');
      expect(loader).not.toBeNull();
    });
  });
});

/** Test component that contains 3 different tagged Buttons. */
@Component({
  selector: 'test-app',
  template: `
    <button
      cui-button
      (click)="onClick()"
      (keyDown)="onClick()"
      [active]="active"
      [attr.aria-label]="ariaLabel"
      [attr.aria-labelledby]="ariaLabelledBy"
      [circle]="circle"
      [class]="class"
      [color]="color"
      [disabled]="disabled"
      [expand]="expand"
      [loading]="loading"
      [removeStyle]="removeStyle"
      [size]="size"
      [type]="type"
    >
      Button
    </button>

    <a
      cui-button
      (click)="onClick()"
      [active]="active"
      [attr.aria-label]="ariaLabel"
      [attr.aria-labelledby]="ariaLabelledBy"
      [circle]="circle"
      [class]="class"
      [color]="color"
      [disabled]="disabled"
      [expand]="expand"
      [href]="href"
      [loading]="loading"
      [removeStyle]="removeStyle"
      [size]="size"
      [type]="type"
      >Anchor</a
    >
    <input
      cui-button
      (click)="onClick()"
      [active]="active"
      [attr.aria-label]="ariaLabel"
      [attr.aria-labelledby]="ariaLabelledBy"
      [circle]="circle"
      [class]="class"
      [color]="color"
      [disabled]="disabled"
      [expand]="expand"
      [href]="href"
      [loading]="loading"
      [removeStyle]="removeStyle"
      [size]="size"
      [type]="type"
    />
  `,
})
export class TestAppComponent {
  clickCount = 0;
  active = false;
  ariaLabel = '';
  ariaLabelledBy = '';
  circle = false;
  class = '';
  color = '';
  disabled = false;
  expand = false;
  href = '';
  loading = false;
  removeStyle = false;
  size = '36';
  styles = '';
  type = 'button';

  onClick() {
    this.clickCount++;
  }
}
// tslint:enable:no-use-before-declare
