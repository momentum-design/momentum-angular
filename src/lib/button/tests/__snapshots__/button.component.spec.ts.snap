// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent of type Anchor should render as an anchor html element with required parameters 1`] = `
<a
  aria-label="aNewAriaLabel"
  aria-labelledby="aNewArialabelledBy"
  class="cui-button cui-button--36"
  cui-button=""
  ng-reflect-active="false"
  ng-reflect-circle="false"
  ng-reflect-class=""
  ng-reflect-color=""
  ng-reflect-disabled="false"
  ng-reflect-expand="false"
  ng-reflect-href=""
  ng-reflect-loading="false"
  ng-reflect-remove-style="false"
  ng-reflect-size="36"
  ng-reflect-type="button"
  type="button"
  width=""
>
  <div>
    
    <span
      ng-reflect-ng-style="[object Object]"
      style="opacity: 1;"
    >
      Anchor
    </span>
  </div>
</a>
`;

exports[`ButtonComponent of type Button should render as a button html element with required parameters 1`] = `
<button
  aria-label="aNewAriaLabel"
  aria-labelledby="aNewArialabelledBy"
  class="cui-button cui-button--36"
  cui-button=""
  ng-reflect-active="false"
  ng-reflect-circle="false"
  ng-reflect-class=""
  ng-reflect-color=""
  ng-reflect-disabled="false"
  ng-reflect-expand="false"
  ng-reflect-loading="false"
  ng-reflect-remove-style="false"
  ng-reflect-size="36"
  ng-reflect-type="button"
  type="button"
  width=""
>
  <div>
    
    <span
      ng-reflect-ng-style="[object Object]"
      style="opacity: 1;"
    >
       Button 
    </span>
  </div>
</button>
`;

exports[`ButtonComponent of type Input should render as an anchor html element with required parameters 1`] = `
<input
  aria-label="aNewAriaLabel"
  aria-labelledby="aNewArialabelledBy"
  class="cui-button cui-button--36"
  cui-button=""
  ng-reflect-active="false"
  ng-reflect-circle="false"
  ng-reflect-class=""
  ng-reflect-color=""
  ng-reflect-disabled="false"
  ng-reflect-expand="false"
  ng-reflect-href=""
  ng-reflect-loading="false"
  ng-reflect-remove-style="false"
  ng-reflect-size="36"
  ng-reflect-type="button"
  type="button"
  width=""
>
  <div>
    
    <span
      ng-reflect-ng-style="[object Object]"
      style="opacity: 1;"
    />
  </div>
</input>
`;
